src/lib/idb.ts

```diff
@@ -165,140 +167,39 @@ export default class IDBStorage {
 
   public delete(entryName: string | string[]): Promise<void> {
     //return Promise.resolve();
-    return this.openDatabase().then((db) => {
-      return new Promise((resolve, reject) => {
-        try {
-          //this.log('delete: `' + entryName + '`');
-          const transaction = db.transaction([this.storeName], 'readwrite');
-          const objectStore = transaction.objectStore(this.storeName);
-
-          transaction.onerror = (e) => {
-            reject(transaction.error);
-            clearTimeout(timeout);
-          };
-
-          transaction.oncomplete = (e) => {
-            this.log('delete: transaction complete', entryName);
-            resolve();
-            clearTimeout(timeout);
-          };
-
-          const timeout = setTimeout(() => {
-            this.log.error('delete: transaction not finished', entryName, transaction);
-          }, 10000);
-
-          if(!Array.isArray(entryName)) {
-            entryName = [].concat(entryName);
-          }
+    if(!Array.isArray(entryName)) {
+      entryName = [].concat(entryName);
+    }
 
-          for(let i = 0, length = entryName.length; i < length; ++i) {
-            const request = objectStore.delete(entryName[i]);
-            request.onerror = (error) => {
-              reject(transaction.error);
-              clearTimeout(timeout);
-            };
-          }
-        } catch(error) {
-          reject(error);
-        }
-      });
-    });
+    return this.getObjectStore('readwrite', (objectStore) => {
+      return (entryName as string[]).map((entryName) => objectStore.delete(entryName));
+    }, DEBUG ? 'delete: ' + entryName.join(', ') : '');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MikeOwino/tweb/pullRequest?prid=7329411)
```suggestion
    }, DEBUG ? "delete: " + entryName.join(', ') : '');
```